www/source/partials/docs/_using-builder-origin-keys.html.md.erb

```diff
@@ -0,0 +1,129 @@
+
+You can create origin keys from your host machine or from the Habitat Studio. 
+
+You can create an origin key pair during your initial Habitat installation by following the instructions in the `hab cli setup` interaction.
+If you've already installed Habitat, you can create keys for additional origins by running `hab origin key generate <ORIGIN>`, where <ORIGIN> is the name for your new Habitat origin,  from either the host machine or from within the studio.
+
+To create origin keys from the command line, use:
+
+```bash
+hab origin key generate <ORIGIN>
+```
+
+Your keys will be stored in `~/.hab/cache/keys` on Linux sytems. 
+
+```bash
+ls ~/.hab/cache/keys
+
+test-origin-20190416223046.pub
+test-origin-20190416223046.sig.key
+```
+
+In this case, the origin is named "test-origin". The string of numbers, "20190416223046" show the date and time the key was created, in this case, 2019-04-16-22-30-46. The public key has the file extension `.pub` and the private key has `.sig.key`, which stands for "signing key", another name for the private key.
+
+### Download origin keys from Builder
+
+To download your private or public origin key, select the download icon from the right end of the key details, under the "Actions" heading.
+
+<img src="/images/screenshots/origin-key-download.png">
+
+To get your keys from Builder from the command line, use the command
+
+```shell
+hab origin key download <ORIGIN>
+```
+
+### Upload origin keys to Builder
+
+When you create Habitat origin keys on your workstation or in the Studio with `hab origin key generate <ORIGIN>`, your origin key pairs are stored in that environment. Habitat Builder cannot access origin keys stored on your workstation or in the Studio, which means that you need to upload keys to Builder either to upload or download your Habitat origins.
+
+At the very least, Builder requires at the public origin key to verify the integrity of your origins, so you'll need to upload it. Builder requires the public origin key to upload artifacts for that origin. You can also upload your private origin key.
+
+Upload origin keys through the Builder, or by using the `hab origin key upload` command. Uploading origin keys uses the Habitat access token for authentication.
+
+#### Upload command
+
+The simplest method for uploading your keys is to use the commands:
+
+```bash
+hab origin key upload <ORIGIN>
+hab origin key upload --secret <ORIGIN>
+```
+
+To upload both origin keys at the same time, use: 
+
+```bash
+hab origin key upload  --secfile <PATH_TO_PRIVATE_KEY> --pubfile <PATH_TO_PUBLIC_KEY>
+```
+
+#### Import command
+
+Use `hab origin key import` to read the key from a standard input stream:
+
+```bash
+hab origin key import <enter or paste key>
+hab origin key import <PATH_TO_KEY>
+curl <URL_THAT_RETURNS_KEY> | hab origin key import
+```
+
+In some Linux-based system, such as MacOS, you may encounter an upload failure.
```

Comment:
macOS isn't a Linux-based system, but the `SSL_CERT_FILE` issue *is* macOS specific, so we can just say that.